src/osd/OSD.cc

```diff
@@ -2772,8 +2772,6 @@ int OSD::shutdown()
   hb_front_server_messenger->shutdown();
   hb_back_server_messenger->shutdown();
 
-  peering_wq.clear();
-
   return r;
```

Comment:
I apparently introduced that in 1493e7dbfb4accd931e27c563526510492a11a8b, but did not mention why.  I think it's because it's possible for other threads to put something into the peering_wq after the threadpool is stopped.  I suspect that this is actually important.